documentation-website/Writerside/topics/DAO-CRUD-Operations.topic

```diff
@@ -0,0 +1,254 @@
+            <code>Expression</code>. This can be done using the <code>wrapAsExpression()</code>
+            function:</p>
+        <code-block lang="kotlin">
+            val expression = wrapAsExpression&lt;Int&gt;(Users
+                .select(Users.id.count())
+                .where {
+                    Cities.id eq Users.cityId
+                })
+            val cities = Cities
+                .selectAll()
+                .orderBy(expression, SortOrder.DESC)
+                .toList()
+        </code-block>
+    </chapter>
+
+    <chapter title="Add computed fields to entity class" id="add-computed-fields-to-entity-class">
+        <p>Suppose that you want to use a window function to rank films with each entity fetch. The companion object of
+            the entity class can override any open
+            function in <code>EntityClass</code>. However, to achieve this functionality, you only need to override
+            <code>searchQuery()</code>. The results of the function can then be accessed through a property of the entity class:</p>
+
+        <code-block lang="kotlin"><![CDATA[
+            object StarWarsFilms : IntIdTable() {
+                val sequelId = integer("sequel_id").uniqueIndex()
+                val name = varchar("name", 50)
+                val rating = double("rating")
+
+                val rank = Rank().over().orderBy(rating, SortOrder.DESC)
+            }
+
+            class StarWarsFilm(id: EntityID<Int>) : IntEntity(id) {
+                var sequelId by StarWarsFilms.sequelId
+                var name by StarWarsFilms.name
+                var rating by StarWarsFilms.rating
+
+                val rank: Long
+                    get() = readValues[StarWarsFilms.rank]
+
+                companion object : IntEntityClass<StarWarsFilm>(StarWarsFilms) {
+                    override fun searchQuery(op: Op<Boolean>): Query {
+                        return super.searchQuery(op).adjustSelect {
+                            select(columns + StarWarsFilms.rank).set
+                        }
+                    }
+                }
+            }
+
+            transaction {
+                StarWarsFilm.new {
+                    sequelId = 8
+                    name = "The Last Jedi"
+                    rating = 4.2
+                }
+                // more insertions ...
+                entityCache.clear()
+
+                // fetch entities with value (or store entities then read value)
+                StarWarsFilm.find { StarWarsFilms.name like "The%" }.map { it.name to it.rank }
+            }
+        ]]></code-block>
+    </chapter>
+
+    <chapter title="Auto-fill created and updated columns on entity change" id="auto-fill-created-and-updated-columns-on-entity-change">
+        <p>See example by @PaulMuriithi <a href="https://github.com/PaulMuriithi/ExposedDatesAutoFill/blob/master/src/main/kotlin/app/Models.kt">here</a>.</p>
+    </chapter>
```

Comment:
Outside this PR, if you think it's worth doing, we could rewrite this using our own sample code and avoid having to link externally. Would probably be a good opportunity to include content on table/entity inheritance and interceptors, etc.

Reply:
Yep, we should definitely do something about this. I created a new issue at [EXPOSED-572](https://youtrack.jetbrains.com/issue/EXPOSED-572). If you could help with writing the sample, it would be great 🙏